internal/unionstore/memdb_arena.go

```diff
@@ -353,7 +353,7 @@ func (l *memdbVlog) getSnapshotValue(addr memdbArenaAddr, snap *MemDBCheckpoint)
 	if result.isNull() {
 		return nil, false
 	}
-	return l.getValue(addr), true
```

Comment:
Interesting, we may need to check if there is any caller relying on this mistake behaviour.

Reply:
Yeah, I don't know if the tidb users are affected by this mistake, it should only happens for the queries which write data and do unionscan in the same time.

Reply:
There is, I have a regression test in pingcap/tidb#54929.